src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -1424,41 +1444,97 @@ def cli_cosmosdb_identity_show(client, resource_group_name, account_name):
     return cosmos_db_account.identity
 
 
-def cli_cosmosdb_identity_assign(client,
+def cli_cosmosdb_identity_assign(cmd,
+                                 client,
                                  resource_group_name,
-                                 account_name):
-    """ Show the identity associated with a Cosmos DB account """
+                                 account_name,
+                                 identities=None):
+    """ Update the identities associated with a Cosmos DB account """
 
     existing = client.get(resource_group_name, account_name)
+    UserAssignedIdentitiesValue = cmd.get_models('UserAssignedIdentitiesValue')
+
+    SYSTEM_ID = '[system]'
+    enable_system = identities is None or SYSTEM_ID in identities
+    new_user_identities = []
+    if identities is not None:
+        new_user_identities = [x for x in identities if x != SYSTEM_ID]
+
+    if existing.identity and existing.identity.type == ResourceIdentityType.system_assigned_user_assigned:
+        identity_types = ResourceIdentityType.system_assigned_user_assigned
+    elif existing.identity and existing.identity.type == ResourceIdentityType.system_assigned and new_user_identities:
+        identity_types = ResourceIdentityType.system_assigned_user_assigned
+    elif existing.identity and existing.identity.type == ResourceIdentityType.user_assigned and enable_system:
+        identity_types = ResourceIdentityType.system_assigned_user_assigned
+    elif new_user_identities and enable_system:
+        identity_types = ResourceIdentityType.system_assigned_user_assigned
+    elif new_user_identities:
+        identity_types = ResourceIdentityType.user_assigned
+    else:
+        identity_types = ResourceIdentityType.system_assigned
 
-    if ResourceIdentityType.system_assigned.value in existing.identity.type:
-        return existing.identity
-
-    if existing.identity.type == ResourceIdentityType.user_assigned.value:
-        identity = ManagedServiceIdentity(type=ResourceIdentityType.system_assigned_user_assigned.value)
+    if identity_types in [ResourceIdentityType.system_assigned, ResourceIdentityType.none]:
+        set_identity = ManagedServiceIdentity(type=identity_types.value)
     else:
-        identity = ManagedServiceIdentity(type=ResourceIdentityType.system_assigned.value)
-    params = DatabaseAccountUpdateParameters(identity=identity)
+        set_user_identities = existing.identity.user_assigned_identities or []
+        for identity in new_user_identities:
+            set_user_identities[identity] = UserAssignedIdentitiesValue()
+
+        set_identity = ManagedServiceIdentity(type=identity_types.value, user_identities=set_user_identities)
+
+    params = DatabaseAccountUpdateParameters(identity=set_identity)
     async_cosmos_db_update = client.begin_update(resource_group_name, account_name, params)
```

Comment:
I originally did not add this logic because it is really difficult to follow the code and I didn't think it was valuable enough to be worthwhile. I've added it, but I'm still not sure it's worth the extra complication.

Reply:
Yeah, I think it's worthwhile, thank you!
The update is an operation which takes long time, why not just return directly to customer if there's no update.